e2e/cypress/e2e/sidebar/quickstart-sidebar.cy.ts

```diff
@@ -0,0 +1,415 @@
+        "have.attr",
+        "placeholder",
+        "+ tag 3 selected samples"
+      );
+      cy.get("[data-cy=tagger-switch-label]").click();
+      cy.get("[data-cy=label-tag-input]").should(
+        "have.attr",
+        "placeholder",
+        "+ tag 64 selected labels"
+      );
+    });
+
+    it("When filter exists, sample tag works correctly", () => {
+      cy.get("[data-cy=selected-pill-button]").should("be.visible").click();
+      cy.get("[data-cy=item-action-Clear-selected-samples]")
+        .should("be.visible")
+        .click();
+      cy.get("[data-cy=checkbox-tags]").should("be.visible").click();
+      cy.get("[data-cy=sidebar-field-arrow-tags]").should("be.visible").click();
+
+      // 1. Filter Mode: tag samples in the dataset
+      cy.get("[data-cy=sidebar-field-arrow-predictions]")
+        .should("be.visible")
+        .click();
+      cy.get("[data-cy=sidebar-column]").scrollTo("bottom");
+      cy.contains("1,023", { timeout: 5000 }).should("be.visible");
+      cy.get(
+        "[data-cy=categorical-filter-predictions\\.detections\\.label]"
+      ).should("exist");
+      cy.get('[data-cy="selector-\\+ filter by label"]').should("be.visible");
+      cy.get("[data-cy=selector-div-predictions\\.detections\\.label]").within(
+        () => {
+          cy.get("input").type("carrot").wait(300).type("{enter}");
+        }
+      );
+      cy.get("[data-cy=tagger-pill-button]").click();
+      cy.get("[data-cy=sample-tag-input]")
+        .type("carrot")
+        .wait(200)
+        .type("{enter}");
+      cy.get("[data-cy=Button-Apply").should("be.visible").click();
+      // verify the tag rendered in looker
+      cy.get("[data-cy=tag-tags-carrot]").should("have.length", 7);
+      // verify the sample tag count in the menu
+      cy.get("[data-cy=checkbox-carrot]")
+        .should("be.visible")
+        .first()
+        .within(() => {
+          cy.get("span").contains(7).should("exist");
+          cy.get("span").contains("carrot").should("exist");
+        });
+      cy.get("[data-cy=sidebar-column]").scrollTo("bottom");
+      cy.get("[data-cy=Button-Reset]").should("be.visible").click();
+
+      // 2. Select Mode: select some samples in the grid
+      selectSamplesInGrid([0, 1, 2]);
+      cy.get("[data-cy=tagger-pill-button]").click();
+      cy.get("[data-cy=sample-tag-input]")
+        .type("top3")
+        .wait(30)
+        .type("{enter}");
+      cy.get("[data-cy=Button-Apply").should("be.visible").click();
+      // verify the tag rendered in looker
+      cy.get("[data-cy=tag-tags-top3]").should("have.length", 3);
+      // verify the sample tag count in the menu
+      cy.get("[data-cy=checkbox-top3]")
+        .should("be.visible")
+        .first()
+        .within(() => {
+          cy.get("span").contains(3).should("be.visible");
+          cy.get("span").contains("top3").should("be.visible");
+        });
+      cy.get("[data-cy=refresh-fo]").should("exist").click();
```

Comment:
nit: If the goal is refresh state, I'd do it at the beginning and not the end.

https://docs.cypress.io/guides/references/best-practices#State-reset-should-go-before-each-test